crates/rust-lib/src/lib.rs

```diff
@@ -586,12 +590,20 @@ pub trait RustGenerator<'a> {
                 self.push_str("#[derive(wasmtime::component::Lower)]\n");
                 self.push_str("#[component(record)]\n");
             }
-
+            let mut derives = additional_derives.clone();
             if info.is_copy() {
                 self.push_str("#[repr(C)]\n");
-                self.push_str("#[derive(Copy, Clone)]\n");
+                derives.extend(["Copy", "Clone"].into_iter().map(|s| s.to_string()));
             } else if info.is_clone() {
-                self.push_str("#[derive(Clone)]\n");
+                derives.insert("Clone".to_string());
+            }
+            if !derives.is_empty() {
+                self.push_str("#[derive(");
+                // NOTE(thomastaylor312): Since this is code generation, I don't mind the iteration
+                // and allocation of a new vec here. If it ever matters, we can create a function
+                // that glues it together via iterator
```

Comment:
No worries on leaving a note here, very little of wit-bindgen is optimized for performance :)